File: packages/pontil_core/test/pontil_core_test_ffi.mjs
import { Ok, Error } from "./gleam.mjs";

export function getExitCode() {
  if (globalThis?.Deno) {
    return new Ok(globalThis.Deno.exitCode ?? 0);
  } else if (globalThis?.process) {
    return new Ok(globalThis.process.exitCode ?? 0);
  }
  return new Error(undefined);
}

export function clearExitCode() {
  if (globalThis?.Deno) {
    globalThis.Deno.exitCode = 0;
  } else if (globalThis?.process) {
    globalThis.process.exitCode = 0;
  }
}

File: packages/pontil_core/test/pontil_core_test_ffi.erl
-module(pontil_core_test_ffi).
-export([get_exit_code/0, clear_exit_code/0]).

get_exit_code() ->
    case erlang:get(pontil_process_exit_code) of
        undefined -> {error, nil};
        Code -> {ok, Code}
    end.

clear_exit_code() ->
    erlang:erase(pontil_process_exit_code),
    nil.
